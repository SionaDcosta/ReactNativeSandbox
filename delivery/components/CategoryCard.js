import { View, Text, TouchableOpacity, Image } from 'react-native'
import React from 'react'
import tailwind from 'twrnc'

const CategoryCard = ({imgUrl, title}) => {
  return (
    <TouchableOpacity style={tailwind`relative mr-2`} testID="category-card" >
        <Image
            source={{uri:imgUrl}}
            style={tailwind`h-20 w-20 rounded`}
            testID="category-card-image"
        />
        <Text 
            style={tailwind`absolute bottom-1 left-1 text-white font-bold`}
          testID="category-card-title"
        >{title}</Text>
    </TouchableOpacity>
  )
}

export default CategoryCard