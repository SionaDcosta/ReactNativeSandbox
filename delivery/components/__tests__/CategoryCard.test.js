// CategoryCard.test.js

import React from 'react';
import { render, screen } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import CategoryCard from '../CategoryCard';

describe('CategoryCard Component', () => {
  const imgUrl = 'https://example.com/image.jpg';
  const title = 'Example Title';

  it('renders correctly', () => {
    const tree = renderer.create(<CategoryCard imgUrl={imgUrl} title={title} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders CategoryCard component', () => {
    render(<CategoryCard imgUrl={imgUrl} title={title} />);
    // Check if the component renders
    expect(screen.getByTestId('category-card')).toBeDefined();
    
    // Check if the image is rendered with correct source
    const imageElement = screen.getByTestId('category-card-image');
    expect(imageElement).toBeDefined();
    expect(imageElement.props.source.uri).toBe(imgUrl);
    
    // Check if the title text is rendered correctly
    const titleElement = screen.getByTestId('category-card-title');
    expect(titleElement).toBeDefined();
    expect(titleElement.props.children).toBe(title);
  });

  it('renders CategoryCard component without onPress prop', () => {
    render(<CategoryCard imgUrl={imgUrl} title={title} />);

    // Check if the component renders
    expect(screen.getByTestId('category-card')).toBeDefined();
    
    // Check if the image is rendered with correct source
    const imageElement = screen.getByTestId('category-card-image');
    expect(imageElement).toBeDefined();
    expect(imageElement.props.source.uri).toBe(imgUrl);
    
    // Check if the title text is rendered correctly
    const titleElement = screen.getByTestId('category-card-title');
    expect(titleElement).toBeDefined();
    expect(titleElement.props.children).toBe(title);
  });

});
