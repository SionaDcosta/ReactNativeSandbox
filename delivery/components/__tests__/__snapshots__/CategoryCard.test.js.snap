// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCard Component renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "marginRight": 8,
      "opacity": 1,
      "position": "relative",
    }
  }
  testID="category-card"
>
  <Image
    source={
      {
        "uri": "https://example.com/image.jpg",
      }
    }
    style={
      {
        "borderRadius": 4,
        "height": 80,
        "width": 80,
      }
    }
    testID="category-card-image"
  />
  <Text
    style={
      {
        "bottom": 4,
        "color": "#fff",
        "fontWeight": "bold",
        "left": 4,
        "position": "absolute",
      }
    }
    testID="category-card-title"
  >
    Example Title
  </Text>
</View>
`;
